hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequest.java

```diff
@@ -227,7 +227,11 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       long correctedSpace = omKeyInfo.getReplicatedSize();
       // if keyToDelete isn't null, usedNamespace needn't check and
       // increase.
-      if (keyToDelete != null && !omBucketInfo.getIsVersionEnabled()) {
+      if (keyToDelete != null && isHSync) {
```

Comment:
This is okay now when we essentially allocate just one block at a time. The internal logical does permit OM to request more than one block when file is created. I am not sure when that happens, but a full solution would need to somehow record the allocated but uncommitted blocks when hsync. For example, it could be achieved by making a deep copy of the openFileTable entry.